diff --git a/server/src/main/java/com/genymobile/scrcpy/daemon/control/CommandHandler.java b/server/src/main/java/com/genymobile/scrcpy/daemon/control/CommandHandler.java
new file mode 100644
index 00000000..70a161fe
--- /dev/null
+++ b/server/src/main/java/com/genymobile/scrcpy/daemon/control/CommandHandler.java
@@ -0,0 +1,7 @@
+package com.genymobile.scrcpy.daemon.control;
+
+import com.genymobile.scrcpy.control.ControlMessage;
+
+public interface CommandHandler {
+    void handle(ControlMessage msg, CommandContext ctx) throws Exception;
+}
