diff --git a/server/src/main/java/com/genymobile/scrcpy/daemon/control/ExecutionPolicy.java b/server/src/main/java/com/genymobile/scrcpy/daemon/control/ExecutionPolicy.java
new file mode 100644
index 00000000..4ccf0b4e
--- /dev/null
+++ b/server/src/main/java/com/genymobile/scrcpy/daemon/control/ExecutionPolicy.java
@@ -0,0 +1,6 @@
+package com.genymobile.scrcpy.daemon.control;
+
+public enum ExecutionPolicy {
+    FAST,
+    SLOW
+}
