diff --git a/server/src/main/java/com/genymobile/scrcpy/control/ControlMessageReader.java b/server/src/main/java/com/genymobile/scrcpy/control/ControlMessageReader.java
index b2591d6d..5f920de5 100644
--- a/server/src/main/java/com/genymobile/scrcpy/control/ControlMessageReader.java
+++ b/server/src/main/java/com/genymobile/scrcpy/control/ControlMessageReader.java
@@ -1,5 +1,6 @@
 package com.genymobile.scrcpy.control;
 
+import com.genymobile.scrcpy.daemon.control.DaemonControlMessageReader;
 import com.genymobile.scrcpy.model.Position;
 import com.genymobile.scrcpy.util.Binary;
 
@@ -65,6 +66,10 @@ public class ControlMessageReader {
             case ControlMessage.TYPE_SCAN_FILE:
                 return parseScanFile();
             default:
+                ControlMessage daemonMsg = DaemonControlMessageReader.read(type, dis);
+                if (daemonMsg != null) {
+                    return daemonMsg;
+                }
                 throw new ControlProtocolException("Unknown event type: " + type);
         }
     }
